src/main/java/io/sdkman/broker/download/CandidateDownloadHandler.java

```diff
@@ -47,12 +58,44 @@ public void handle(Context ctx) {
                                             downloadResolver.resolve(downloads, p.name())
                                                     .ifPresentOrElse(v -> {
                                                         audit(details, p.id(), v.platform());
+                                                        writeChecksumHeaders(ctx, v.checksums());
```

Comment:
One way you could get around the option checks on line 70 as well as the `.get()` on 93 is by mapping over the option here. In other words, something like the following pseudo-code:
```
v.checksums().forEach { checksums ->
    writeChecksumHeaders(ctx, checksums);
}
```

Reply:
It look cleaner indeed